src/main/java/org/elasticsearch/search/aggregations/Aggregator.java

```diff
@@ -225,7 +211,17 @@ public void gatherAnalysis(BucketAnalysisCollector results, long bucketOrdinal)
         };
     }
 
-    protected void preCollection() {
+    public Map<String, Object> getMetaData() {
```

Comment:
All other getters are named X() rather than getX() so should this be named metadata() ?